sbi/inference/snpe/snpe_base.py

```diff
@@ -86,11 +86,16 @@ def __init__(
             density_estimator = utils.posterior_nn(
                 model="maf", prior=self._prior, x_o=self._x_o,
             )
+        # else: check density estimator for valid prior etc.
+        # XXX: here, the user could sneak in an invalid prior and x_o by providing a
+        # density estimator with invalid .prior and .x_o. This is bypassing
+        # the input checks.
+
```

Comment:
Good catch, that is a problem indeed. Another solution would be to disallow the user from specifying a density estimator. Then, the user would only pass either
a) Nothing ==> default density estimator
b) A dictionary with keys `"model"`, `"hidden_units"`, `"num_blocks"` etc. ==> used to initialize the density estimator.

I don't have a strong opinion on whether we should do this, but it would allow us to get rid of the described problem without checking for a valid prior twice.

Reply:
Yes, I like that option. I think it makes it easier for the user as she doesnt have to look in `get_nn_models` for how to build the density estimator.

Reply:
we should do this in a separate PR. this one here just fixes the checks.

Reply:
sure!

Reply:
this is now here #135